starlark/eval.go

```diff
@@ -92,6 +93,10 @@ func (thread *Thread) Uncancel() {
 	atomic.StorePointer((*unsafe.Pointer)(unsafe.Pointer(&thread.cancelReason)), nil)
 }
 
+var (
+	ErrExecutionCancelled = errors.New("Starlark computation cancelled")
```

Comment:
Let's call it just ErrCanceled (note US spelling), and document it:
```
// When a Starlark function call is canceled by a call to
// [Thread.Cancel] or [Thread.CancelWithError],
// the EvalError returned by [Call] will respond to
// errors.Is(err, ErrCanceled).
```

Reply:
I must confess that I wrote "cancelled" because the message says "cancelled". I don't think we want to change the message, as I'm sure that people are string matching at this point... but tell me if you think otherwise.

Reply:
I think we can afford to change it. We never promised any spelling, and if this causes people to have to update their tests, they might also switch to the more robust approach of using `errors.Is(ErrCanceled, err)`.

Reply:
Done.